libcaf_net/caf/net/write_packet_decorator.hpp

```diff
@@ -26,34 +29,44 @@ namespace net {
 template <class Object, class Parent>
 class write_packet_decorator {
 public:
-  dispatcher(Object& object, Parent& parent)
+  // -- member types -----------------------------------------------------------
+
+  using transport_type = typename Parent::transport_type;
+
+  using application_type = typename Parent::application_type;
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  write_packet_decorator(Object& object, Parent& parent)
     : object_(object), parent_(parent) {
     // nop
   }
 
-  template <class Header>
-  void write_packet(const Header& header, span<const byte> payload) {
-    object_.write_packet(parent_, header, payload);
-  }
+  // -- properties -------------------------------------------------------------
 
   actor_system& system() {
-    parent_.system();
+    return parent_.system();
   }
 
-  void cancel_timeout(atom_value type, uint64_t id) {
-    parent_.cancel_timeout(type, id);
+  transport_type& transport() {
+    return parent_.transport();
   }
 
-  void set_timeout(timestamp tout, atom_value type, uint64_t id) {
-    parent_.set_timeout(tout, type, id);
+  // -- member functions -------------------------------------------------------
+
+  template <class Header, class... Ts>
+  void write_packet(const Header& header, span<const byte> payload,
```

Comment:
We always pass the header as byte span:

```suggestion
  void write_packet(span<const byte> header, span<const byte> payload,
```